go/ql/lib/semmle/go/security/TaintedPathCustomizations.qll

```diff
@@ -93,6 +93,23 @@ module TaintedPath {
     }
   }
 
+  /**
+   * A call to `mux.Vars(path)`, considered to sanitize `path` against path traversal.
+   * Only enabled when `SkipClean` is not set true.
+   */
+  class MuxVarsSanitizer extends Sanitizer {
+    MuxVarsSanitizer() {
+      exists(Function m |
+        m.hasQualifiedName(package("github.com/gorilla/mux", ""), "Vars") and
+        this = m.getACall().getResult()
+      ) and
+      not exists(CallExpr f |
+        f.getTarget().hasQualifiedName(package("github.com/gorilla/mux", ""), "SkipClean") and
```

Comment:
```suggestion
        f.getTarget().(Method).hasQualifiedName(package("github.com/gorilla/mux", ""), "Router", "SkipClean") and
```
This mistake highlights that there isn't a test for this functionality. Please add one.

Reply:
I am having trouble adding both true and false. Due to our limited gorilla mux support, I wrote the sanitizer so that if `SkipClean(true)` exists, then all Vars accesses are considered valid path traversal sources. This means adding `SkipClean(false)` to another file (in the tests directory) doesn't do anything since they are in the same database.

Reply:
Having thought about it, I think you should have two extra test directories: one with a call to `Router.SkipClean(true)` and one with a call to `Router.SkipClean(false)`. And of course each should contain a copy of the test function you've added in this PR.